apiserver/facades/client/applicationoffers/applicationoffers.go

```diff
@@ -91,7 +91,7 @@ func NewOffersAPI(ctx facade.Context) (*OffersAPI, error) {
 	return createOffersAPI(
 		GetApplicationOffers,
 		environFromModel,
-		GetStateAccess(ctx.State()),
+		GetStateAccess(ctx.State(), ctx.StatePool()),
```

Comment:
I had missed that a pool was being used separately here. I would actually like to update the PR to extend the Backend interface to have a GetBackend method and not pass the pool directly to createOffersAPI at all.

Happy to discuss further of course.